hoomd/snapshot.py

```diff
@@ -109,3 +109,38 @@ def replicate(self, nx, ny, nz):
 
     def _broadcast_box(self):
         self._cpp_obj._broadcast_box(self._comm.cpp_mpi_conf)
+
+    @classmethod
+    def _from_gsd_snapshot(cls, gsd_snapshot):
+        snap = cls()
+        # Set all particle attributes in snap from gsd_snapshot
+        for key in vars(gsd_snapshot.particles):
+            val = vars(gsd_snapshot.particles)[key]
+            if val is not None:
+                try:
+                    setattr(snap.particles, key, val)
+                except AttributeError:
+                    try:
+                        # Some attributes exist but aren't setable without [:]
+                        x = getattr(snap.particles,key)
+                        x[:] = val
+                    except AttributeError:
+                        # The attribute doesn't exist in hoomd.Snapshot
+                        pass
+
+        # Set all bond attributes
+        for key in vars(gsd_snapshot.bonds):
+            val = vars(gsd_snapshot.bonds)[key]
+            if val is not None:
+                try:
+                    setattr(snap.bonds, key, val)
+                except AttributeError:
+                    try:
+                        x = getattr(snap.bonds,key)
+                        x[:] = val
+                    except AttributeError:
+                        pass
+
+        # Set box attribute
+        snap.configuration.box = gsd_snapshot.configuration.box
```

Comment:
Boxes work slightly differently in gsd and hoomd. Specifically, you need to check if `gsd_snapshot.configuration.dimensions == 2` and if it is force `box[3]=0`.